src/redux/books/BookSlice.js

```diff
@@ -1,30 +1,32 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import { fetchBooks, addBook, removeBook } from '../../API/Api';
 
-const initialState = [
-  {
-    item_id: 'item1',
-    title: 'The Great Gatsby',
-    author: 'John Smith',
-    category: 'Fiction',
-  },
-  {
-    item_id: 'item2',
-    title: 'Anna Karenina',
-    author: 'Leo Tolstoy',
-    category: 'Fiction',
-  },
-  {
-    item_id: 'item3',
-    title: 'The Selfish Gene',
-    author: 'Richard Dawkins',
-    category: 'Nonfiction',
-  },
-];
+const initialState = {
+  books: [],
+  loading: false,
+  error: '',
+};
+
+export const fetchBooksAsync = createAsyncThunk('books/fetchBooks', async () => {
+  const books = await fetchBooks();
+  return books;
+});
+
+export const addBookAsync = createAsyncThunk('books/addBook', async (book) => {
+  const newBook = await addBook(book);
+  return newBook;
+});
+
+export const removeBookAsync = createAsyncThunk('books/removeBook', async (bookId) => {
+  await removeBook(bookId);
+  return bookId;
+});
 
 const booksSlice = createSlice({
   name: 'books',
   initialState,
   reducers: {
+    /*
```

Comment:
Please remove all commented out code from your project to keep it clean🙏

Reply:
This change has been done well